### PR TITLE
TIS21-2521

### PR DESCRIPTION
**Pull request by EdwardBarclay**
refactor: alter delete assessment

This will need to be pushed with the Admins-UI PR for TIS21-2521

### CODE REVIEW

**Comment by hihilary** [comment]
I found there's another api for hard deletion: `@DeleteMapping("/{traineeId}/assessments/{assessmentId}")`(https://github.com/Health-Education-England/TIS-ASSESSMENTS/blob/b3ca2416f94579c800ef22e4b66f73741e13c3ba/assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResource.java#L249), so wondering which one is better to choose.

File: assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResource.java [hunk 1]
```diff
@@ -254,47 +254,12 @@ public AssessmentResource(AssessmentService assessmentService) {
...
-  /**
-   * DELETE  /assessments/:assessmentId : delete a specific assessment.
-   *
-   * @param assessmentId the id of the assessment to delete
-   * @return the ResponseEntity with status 200 (OK)
-   */
-  @DeleteMapping("/assessments/{assessmentId}")
-  @Timed
-  @PreAuthorize("hasAuthority('assessment:add:modify:entities')")
-  public ResponseEntity<Void> deleteAssessment(@PathVariable Long assessmentId) {
```
Comment (hihilary) [inline assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResource.java:266]: This API is used in bulk upload, so should not be deleted.

**Review by hihilary — CHANGES_REQUESTED** [review]

File: assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java [hunk 2]
```diff
@@ -257,49 +257,30 @@ public AssessmentDTO findOne(Long id) {
 
   @Override
   @Transactional
-  public boolean deleteTraineeAssessment(Long assessmentId, Long traineeId) {
-    Preconditions.checkNotNull(traineeId);
+  public boolean deleteAssessment(Long assessmentId) {
```
Comment (hihilary) [inline assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java:260]: Seems there're no changes for `deleteAssessment` and `deleteTraineeAssessment`. But on this page they look messy to me.

File: assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResource.java [hunk 3]
```diff
@@ -264,10 +264,10 @@ public AssessmentResource(AssessmentService assessmentService) {
...
   @PreAuthorize("hasAuthority('assessment:add:modify:entities')")
   public ResponseEntity<Void> deleteAssessment(@PathVariable Long assessmentId) {
-    assessmentService.deleteAssessment(assessmentId);
-    return ResponseEntity.ok()
-        .headers(HeaderUtil.createEntityDeletionAlert(ENTITY_NAME, assessmentId.toString()))
-        .build();
+
+    boolean success =  assessmentService.deleteAssessment(assessmentId);
+
+    return new ResponseEntity<>(success ? HttpStatus.OK : HttpStatus.BAD_REQUEST);
```
Comment (hihilary) [inline assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResource.java:270]: No tests for the unhappy path.
Is there any benefit for this change?

File: assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java [hunk 4]
```diff
@@ -97,20 +97,6 @@ public void setup() {
...
 
-  @Test
-  public void deleteTraineeAssessmentShouldReturnBadRequestWhenAssessmentForTraineeNotFound() throws Exception {
-    when(assessmentServiceMock.deleteTraineeAssessment(ASSESSMENT_ID, TRAINEE_ID)).thenReturn(false);
-    mockMvc.perform(delete("/api/trainee/{traineeId}/assessments/{assessmentId}", TRAINEE_ID, ASSESSMENT_ID))
-        .andExpect(status().isBadRequest());
-  }
-
-  @Test
-  public void deleteTraineeAssessmentShouldReturnOKWhenAssessmentForTraineeIsFound() throws Exception {
```
Comment (hihilary) [inline assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java:108]: This test is not for soft-delete API.

File: assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java [hunk 5]
```diff
@@ -466,26 +452,6 @@ public void updateSpecificTraineeAssessmentShouldBadRequestWhenAssessmentIdDoesn
...
-
-    when(assessmentServiceMock.deleteTraineeAssessment(ASSESSMENT_ID_1, TRAINEE_ID)).thenReturn(false);
-
-    mockMvc.perform(delete("/api/trainee/{traineeId}/assessments/{assessmentId}", TRAINEE_ID, ASSESSMENT_ID_1))
-        .andExpect(status().isBadRequest());
-
-  }
-
-  @Test
-  public void deleteTraineeAssessmentShouldReturnOkAfterSuccessfulDeletion() throws Exception {
```
Comment (hihilary) [inline assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java:480]: This test is not for soft-delete API.

**Review by hihilary — CHANGES_REQUESTED** [review]

File: assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java (hunk 2 above)
Comment (EdwardBarclay) [reply assessments-service/src/main/java/com/transformuk/hee/tis/assessment/service/service/impl/AssessmentServiceImpl.java:260]: the deleteAssessment and deleteTraineeAssessment have both remained the same they appear to have been shuffled during the process, the front end has been routed from softDeleteTraineeAssessment which has been deleted to deleteTraineeAssessment and deleteAssessment is used by another service

File: assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java [hunk 6]
```diff
@@ -493,6 +493,20 @@ public void deleteAssessmentShouldReturnOK() throws Exception {
...
+        .andExpect(status().isBadRequest());
+  }
+
+//  @Test
+//  public void deleteAssessmentShouldReturnError() throws Exception {
+//    when(assessmentServiceMock.deleteAssessment(ASSESSMENT_ID)).thenReturn(true);
+//    mockMvc.perform(delete("/api/trainee/assessments/{assessmentId}", ASSESSMENT_ID))
+//        .andExpect(status().);
+//  }
+
```
Comment (hihilary) [inline assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java:509]: The comments above need to be removed.

Comment (EdwardBarclay) [reply assessments-service/src/test/java/com/transformuk/hee/tis/assessment/service/api/AssessmentResourceTest.java:509]: ahh completely forgot that I had commented that out thanks!

**Pull request merged**